prov/verbs/src/fi_verbs.h

```diff
@@ -266,7 +266,8 @@ typedef int (*fi_ibv_trywait_func)(struct fid *fid);
 /* The number of valid OFI indexer bits in the connection key used during
  * XRC connection establishment. Note that only the lower 32-bits of the
  * key are exchanged, so this value must be kept below 32-bits. */
-#define VERBS_TAG_INDEX_BITS	18
+#define VERBS_CONN_TAG_INDEX_BITS	18
+#define VERBS_CONN_TAG_INVALID		0xFFFFFFFF
```

Comment:
What is the association between these 2 defines, if any?

Reply:
CONN_TAG_INDEX_BITS defines the number of bits in the connection tag that will be used to map a tag to a EP. The CONN_TAG_INVALID is used to identify if a tag is not valid. The assumption I was going with here is that there will not be 256K - 1 simultaneous connections all in progress from the same process (so not all of the 18 tag bits would ever be set).

Reply:
Why isn't CONN_TAG_INVALID the same length as TAG_INDEX_BITS?

Reply:
The connection tag itself is 32-bits, although only 18-bits are meaningful to the mapping.  it just made more sense to me to set the entire field since the entire field is used in comparisons (i.e. extra key sequence bits are included). It could be changed to use only the lower 18-bits without changing the behavior.

Reply:
To someone within knowledge of what's going on (i.e. me), the reason for only using 18 out of 32 bits is odd.   The selection of 18 is odd.  And seeing a separate define that covers the same 18 bits, but defines the full 32 is extra odd.  Having some sort of comment or better define to indicate what these are would be nice.  And having consistency between the defines would be ideal.

Reply:
This is all based on using the ofi_key_idx logic to generate unique keys to be passed to remote peers to refer back to local objects.  The key as defined by OFI header comments, is some number of high order sequence bits, and some number of unique low order bits (the latter referred to as the index bits). The extra upper bits provide a little bit of extra ability to detect use of a stale key. 18-bits was chosen to allow a large number of unique tags to exist, while still having a window to detect a stale tag was returned from a peer.

I limit the key to the low order 32-bits and state in the comments that we must keep the unique low order bits less than 32, to reduce the amount of CM data required for the exchange. Maybe it would have been better to have not called it a "conn_tag", but instead of used "conn_key", but "tag" was a design notation, and the ofi_key_idx ended up being an implementation.

I will address your concerns and update the PR, but if you feel there is something wrong with the above logic we can discuss offline. Thanks.

Reply:
I don't disagree with the logic.  I would keep the defines as-is.  Renaming tag to key seems useful, since tag means something else in libfabric.  But I don't think that's the 'issue'.  What I was missing is that the full key is 32-bits.  Simply updating the comments would likely help.  I would change "only the lower 32 bits of the key" to instead define the key as 32-bits, comprised of the 2 fields that you call out above.  thanks

Reply:
I have reworked the comments with your recommendations, and agree that changing "conn_tag" to "conn_key" throughout the code would remove one area of possible confusion. The latter will touch a bunch of files, so I will defer that for a subsequent PR. Thanks again for the comments.

Reply:
thanks!  looks good